pytorch_forecasting/data/timeseries.py

```diff
@@ -418,56 +517,61 @@ def __init__(
                 randomize_length = None
             else:
                 randomize_length = (0.2, 0.05)
+
         self.randomize_length = randomize_length
+
         if min_prediction_idx is None:
             min_prediction_idx = data[self.time_idx].min()
         self.min_prediction_idx = min_prediction_idx
+
         self.constant_fill_strategy = constant_fill_strategy
-        self._constant_fill_strategy = (
-            {} if constant_fill_strategy is None else deepcopy(constant_fill_strategy)
-        )
+        self._constant_fill_strategy = _coerce_to_dict(constant_fill_strategy)
+
         self.predict_mode = predict_mode
         self.allow_missing_timesteps = allow_missing_timesteps
         self.target_normalizer = target_normalizer
+
         self.categorical_encoders = categorical_encoders
-        self._categorical_encoders = (
-            {} if categorical_encoders is None else deepcopy(categorical_encoders)
-        )
+        self._categorical_encoders = _coerce_to_dict(categorical_encoders)
+
         self.scalers = scalers
-        self._scalers = {} if scalers is None else deepcopy(scalers)
+        self._scalers = _coerce_to_dict(scalers)
+
         self.add_target_scales = add_target_scales
         self.variable_groups = variable_groups
-        self._variable_groups = (
-            {} if variable_groups is None else deepcopy(variable_groups)
-        )
+        self._variable_groups = _coerce_to_dict(variable_groups)
+
         self.lags = lags
-        self._lags = {} if lags is None else deepcopy(lags)
+        self._lags = _coerce_to_dict(lags)
 
         # add_encoder_length
         if isinstance(add_encoder_length, str):
-            assert (
-                add_encoder_length == "auto"
-            ), f"Only 'auto' allowed for add_encoder_length but found {add_encoder_length}"
+            msg = (
```

Comment:
Somehow the check for bool instance was ommited here!

Reply:
can you explain? Not sure if I get what you mean.

Reply:
It is a minor missing conditional check for bool. It shows that `add_encoder_length` can be either `bool` or `"auto"` but only the `str` instance is being checked here.

Reply:
oh, I see. The `bool` check is in `_check_params` now, and the `str` check I left there because it has a check fused to a parameter change (which we should later address, but that would be change to logic)